tests/components/garmin_connect/test_config_flow.py

```diff
@@ -0,0 +1,113 @@
+"""Test the Garmin Connect config flow."""
+from unittest.mock import patch
+
+from garminconnect import (
+    GarminConnectAuthenticationError,
+    GarminConnectConnectionError,
+    GarminConnectTooManyRequestsError,
+)
+import pytest
+
+from homeassistant import data_entry_flow
+from homeassistant.components.garmin_connect.const import DOMAIN
+from homeassistant.const import CONF_ID, CONF_PASSWORD, CONF_USERNAME
+
+from tests.common import MockConfigEntry
+
+MOCK_CONF = {
+    CONF_ID: "First Lastname",
+    CONF_USERNAME: "my@email.address",
+    CONF_PASSWORD: "mypassw0rd",
+}
+
+
+@pytest.fixture(name="mock_garmin_connect")
+def mock_garmin():
+    """Mock Garmin."""
+    with patch("homeassistant.components.garmin_connect.config_flow.Garmin",) as garmin:
+        garmin.return_value.get_full_name.return_value = MOCK_CONF[CONF_ID]
+        yield garmin
+
+
+async def test_show_form(hass):
+    """Test that the form is served with no input."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": "user"}
+    )
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "user"
+
+
+async def test_step_user(hass, mock_garmin_connect):
+    """Test registering an integration and finishing flow works."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": "user"},
+        data={"username": "my@email.address", "password": "mypassw0rd"},
+    )
+    assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
+    assert result["data"] == MOCK_CONF
+
+
+async def test_connection_error(hass):
+    """Test for connection error."""
+    with patch(
+        "homeassistant.components.garmin_connect.config_flow.Garmin.login",
+        side_effect=GarminConnectConnectionError("errormsg"),
```

Comment:
Remove the patch and do this instead.
```py
mock_garmin_connect.login.side_effect = GarminConnectConnectionError("errormsg")
```

Reply:
I tried that, but then the mock fails, it gets the website.
Will have another try/look. So close...

Reply:
Did you change the fixture yet? I bumped my comment above.

Reply:
Change and seem to have fixed it, thanks a lot!
This integration is now equally written then most in core as far as I can see... 
Who determines Quality Scales? :smile:
I guess the CI test still fails because of an issue in another parts of the branch.